corporate/lib/stripe.py

```diff
@@ -1412,8 +1422,10 @@ def process_initial_upgrade(
         should_schedule_upgrade_for_legacy_remote_server: bool = False,
     ) -> None:
         is_self_hosted_billing = not isinstance(self, RealmBillingSession)
-        customer = self.update_or_create_stripe_customer()
-        assert customer.stripe_customer_id is not None  # for mypy
+        if charge_automatically:
+            customer = self.update_or_create_stripe_customer()
+        else:
+            customer = self.update_or_create_customer()
```

Comment:
> Since we are no longer creating an invoice here for non charge_automatically plans

I took a slightly different approach in the updated code. Now, a boolean `stripe_invoice_payment_succeeded` will decide whether to create an invoice or not.

We still have code for cloud customers where for "Pay by invoice" billing modality, it immediately upgrades them and generates an invoice. This is not possible by UI. But this would be a separate project altogether to decide how to deal with customers who want to upgrade to various plans with "Pay by invoice".

So, the current code offers three possibility:
* Charge automatically
* Manually send invoice for fixed price plan & customer automatically upgrades as soon as they pay.
* Choose billing_modality = "Pay by invoice", we upgrade instantly & invoice is sent. (Backend code is available but no way to select "Pay by invoice" in /upgrade.)